File: 230207-dojo/javascript/solution.js
// 시간 초과 코드(마지막 4개 테스트 통과 못함)

// export default function solution(numbers) {
//   const answer = [];

//   for (let i = 0; i < numbers.length; i += 1) {
//     const standard = numbers[i];
//     let isBiggerNumberExist = false;
//     let j = i + 1;

//     while (j < numbers.length) {
//       if (standard < numbers[j]) {
//         answer.push(numbers[j]);
//         isBiggerNumberExist = true;
//         break;
//       }

//       j += 1;
//     }

//     if (isBiggerNumberExist === false) {
//       answer.push(-1);
//     }
//   }

//   return answer;
// }

File: 230207-dojo/java/app/src/main/java/Solution.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int[] answer = new int[arr.length];
        Stack<Integer> stack = new Stack<>();

        for (int i = arr.length - 1; i > 0; i =- 1) {
            while (!stack.isEmpty()) {
                if (stack.peek() > arr[i]) {
                    answer[i] = stack.peek();
                    break;
                } else {
                    stack.pop();
                }
            }
            if (stack.isEmpty()) {
                answer[i] = -1;
            }
            stack.push(arr[i]);
        }
        return answer;
    }
}
